### ISSUE TITLE
Bug: Updating Module Title Twice Does Not Save Title Changes - 5.1.0

### ISSUE THREAD

**Issue by thabaum** [oqtane/oqtane.framework#3977]
# ISSUE

## Description

The module title is not being updated once you set it, and try to reset it, without refreshing the page to get a fresh UI instance.

##  Steps to Reproduce...

1.  Edit page module title in page settings
2. Save changes.
3. While still on that page and most likely still in edit mode try to edit and save the title again
4. Notice the title is still the same and does not get updated.

Happens both static and interactive Oqtane render modes.

I noticed in static at least that the carrot icon becomes inactive as well at times.

![image](https://github.com/oqtane/oqtane.framework/assets/13577556/41bdc9ab-d2fd-4501-a21b-d895a0e896a1)

Not sure if the above issue with inactive dropdown actions is limited to my installation. It is something to be aware of while testing solutions.

**Comment by zyhfish** [comment]
it looks like the data is not synced between different db context instances, need to analyze further to find out the root cause.

**Comment by thabaum** [comment]
could of been fixed by #3979 or a recent update in visual studio. I am not experiencing this issue any longer with today's branch.

**Comment by thabaum** [comment]
@zyhfish hello, I will try to reproduce again.

**Comment by thabaum** [comment]
@zyhfish Hello, I was unable to reproduce this on current dev branch.  Did you test this with a hard refresh of the browser for the site page tested using `Ctrl + F5` as I believe I did?  Maybe this makes the difference?  Thank you.  Cheers!

**Comment by zyhfish** [comment]
Hi @thabaum , it's able to be reproduced in the interactive render mode.

**Comment by zyhfish** [comment]
Hi @thabaum ,
the issue should be caused by #ed7904b67324326d00a86d645ae46ff01c1cb1c6 or #dcc8043cf60751e35e78579e749ca1bd4ce76a60. here is the screencast of the issue:
https://drive.google.com/file/d/1nhFKDMZbZj2rrHU8aGAdaACtDxxAko4i/view?usp=sharing

**Comment by thabaum** [comment]
Closing as fixed in #3988

**Comment by sbwalker** [comment]
reopening as the standard process is that the issue is closed once the PR is merged

**Comment by thabaum** [comment]
I had thought this was fixed again as it was working in interactive mode for me with where I was at testing these other issues, so I switched to  `Static/Server` UI component mode from interactive to verify it works in both modes and the issue could be reproduced.

If the fix is something else other than #3977 this should leave some added clues.  Seems like switching render modes may trigger the issue as well.

**Comment by sbwalker** [comment]
@thabaum  I am unable to reproduce this issue in either Static or Interactive render modes. So I am not going to merge PR #3988 unless the problem can be reliably reproduced.

**Comment by zyhfish** [comment]
Hi @sbwalker , I have attached the reproduce screencast in my previous comment, it can be reproduced with Interactive/Server render mode.

After deeper analyze the code, I have identified the root cause of this issue, it's related to the db context factory changes, we also need to use db context factory in the repository classes as well. please check whether PR #4006 is ok to be merged, after this I will submit another PR to make the same changes in other repository classes as well.

**Comment by sbwalker** [comment]
@zyhfish during the MVP Summit this week I asked the Blazor Product Team what the latest recommendation is for EF Core data access for Blazor (as the guidance has changed many times over the years ie. Scoped, Transient, OwnerComponentBase, etc…). I was told that DbContextFactory is the recommended approach.

So I think we should transition all Repositories to use DbContextFactory. The references to DbContext can be removed from the Repositories. However the service registration for DbContext in Startup must be retained for backward compatibility as custom modules will continue to rely on it (until they transition to DbContextFactory). The module template should be updated to use DbContextFactory as well. Do you have time to make these changes?

The reason why this issue is only coming to the surface now is that it’s a rare combination of events which can trigger it. Basically you need a long-lived component which is doing multiple data access calls to the same DbContext. And the component must be using a server-based rendering approach. So it’s quite rare - however it’s still a problem which can be easily avoided with DbContextFactory.

**Comment by zyhfish** [comment]
Hi @sbwalker , sure, I will work on this and submit a PR later. please check whether #4006 can be merged, or I can include it in the next PR with all other repository classes together.
